generative_models/DCGAN_Tensorflow/main.py

```diff
@@ -0,0 +1,117 @@
+import tensorflow as tf
+import numpy as np
+import os
+import PIL
+from tensorflow.keras.models import Sequential, Model, load_model
+from tensorflow.keras.optimizers import Adam
+from tensorflow.keras.losses import BinaryCrossentropy
+
+from MNIST_model import make_generator,make_discriminator
+from utils.py import plot_loss,generate_and_save_images
+import argparse
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument('--EPOCHS', type = int, default = 50, help = "No of EPOCHS: default 50 ")
+parser.add_argument('--noise_dim', type = int, default = 100, help = "Noise dimension, default 100 ")
+parser.add_argument('--BATCH_SIZE', type = int, default = 128, help = "Batch size, default 128")
+parser.add_argument('--num_examples_to_generate', type = int, default = 16, help = "no of images shown after each epoch in output, default 16")
+
+args = parser.parse_args()
+
+# Loading  MNIST_Dataset
+(train_images, train_labels),(_,_) = tf.keras.datasets.mnist.load_data()
+
+BUFFER_SIZE=60000
+BATCH_SIZE = args.BATCH_SIZE
+EPOCHS = args.EPOCHS
+noise_dim = args.noise_dim
+num_examples_to_generate = args.num_examples_to_generate
+
+seed = tf.random.normal([num_examples_to_generate, noise_dim])
+
+# Preparing and Normalising Dataset
+train_images = train_images.reshape(train_images.shape[0], 28, 28, 1).astype('float32')
+train_images = (train_images - 127.5) / 127.5 
+train_dataset = tf.data.Dataset.from_tensor_slices(train_images).shuffle(BUFFER_SIZE).batch(BATCH_SIZE)
+
+# Making generator and Discriminator
+generator=make_generator(noise_dim)
+discriminator=make_discriminator()
+
+# Defining generator and discriminator losses
+cross_entropy = BinaryCrossentropy(from_logits=True)
+
+def discriminator_loss(real_output, fake_output):
+    real_loss = cross_entropy(tf.ones_like(real_output), real_output)
+    fake_loss = cross_entropy(tf.zeros_like(fake_output), fake_output)
+    total_loss = real_loss + fake_loss
+    return total_loss
+
+def generator_loss(fake_output):
+    return cross_entropy(tf.ones_like(fake_output), fake_output)
+
+# Defining optimizers
+generator_optimizer = Adam(learning_rate=0.0002)  
+discriminator_optimizer = Adam(learning_rate=0.0002)
```

Comment:
add learning rate in parser

Reply:
add as two different lr parameter for gen and dis

Reply:
done